test/integration/test_keyprotect.py

```diff
@@ -0,0 +1,51 @@
+import logging
+import unittest
+import os
+import warnings
+
+import redstone
+from redstone import auth as bxauth
+
+
+logging.basicConfig(level=logging.INFO)
+
+
+class KeyProtectTestCase(unittest.TestCase):
+
+    def setUp(self):
+        # filter out the resource warning message
+        warnings.simplefilter("ignore", ResourceWarning)
+
+        apikey = os.environ.get('IBMCLOUD_API_KEY')
+        tm = bxauth.TokenManager(api_key=apikey)
+        self.rc = redstone.service("ResourceController")
+        self.instance_id, self.crn = self.rc.create_instance(name="test-instance", plan_id=self.rc.KEYPROTECT_PLAN_ID,
+                                                             region="us-south")
+        self.kp = redstone.service("KeyProtect",
+                                   credentials=tm,
+                                   region="us-south",
+                                   service_instance_id=self.instance_id,
+                                   )
+
+    def tearDown(self):
+        self.rc.delete_instance(self.instance_id)
+        self.kp.session.close()
+
+    def test_get_registrations(self):
+        # create a key to be used for test
+        self.key = self.kp.create(name="test-key", root=True)
```

Comment:
Use `self.addCleanup()` for delete

Reply:
This would not work on deleting the key, as addCleanup() happens after tearDown() and in that case the instance is already deleted